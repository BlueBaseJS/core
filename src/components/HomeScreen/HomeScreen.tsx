import { ComponentState } from '../../index';
import React from 'react';

export interface HomeScreenProps {

  /**
   * Used to locate this view in end-to-end tests.
   */
	testID?: string,

}

/**
 * # üè† HomeScreen
 *
 * ## Usage
 * ```jsx
 * <HomeScreen/>
 * ```
 */
export const HomeScreen = (props: HomeScreenProps) =>
(
  <ComponentState
    title="BlueBase"
    description="Welcome to BlueBase Framework!"
    imageSource={require('../../../assets/common/logo.png')}
    styles={{ image: { width: 100, height: 100 } }}
    {...props}
  />
);

HomeScreen.displayName = 'HomeScreen';