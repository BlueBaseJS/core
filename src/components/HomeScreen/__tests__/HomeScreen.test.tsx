import { BlueBaseApp } from '../../../';
import { HomeScreen } from '../HomeScreen';
import React from 'react';
import TestRenderer from 'react-test-renderer';

// beforeEach(() => {
// 	jest.resetModules();
// });

describe('HomeScreen', () => {
	test(`Snapshot HomeScreen`, (done) => {
		const component = TestRenderer.create(
			<BlueBaseApp>
				<HomeScreen/>
			</BlueBaseApp>
		);
		setTimeout(() => {
			const tree = component.toJSON();
			expect(tree).toMatchSnapshot();
			done();
		});
	});

});
