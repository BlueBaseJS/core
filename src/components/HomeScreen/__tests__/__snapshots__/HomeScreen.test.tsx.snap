// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen Snapshot HomeScreen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <View
    style={
      Object {
        "marginBottom": 16,
      }
    }
  >
    <Image
      args={Object {}}
      source="test-file-stub"
      style={
        Array [
          Object {
            "height": 100,
            "width": 100,
          },
          undefined,
        ]
      }
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "fontFamily": undefined,
          "fontSize": 20,
          "fontWeight": "500",
        },
        Array [
          Object {
            "marginBottom": 8,
            "textAlign": "center",
          },
          undefined,
        ],
      ]
    }
  >
    BlueBase
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "rgba(0, 0, 0, 0.87)",
          "fontFamily": undefined,
          "fontSize": 14,
          "fontWeight": "400",
        },
        Array [
          Object {
            "marginBottom": 8,
            "textAlign": "center",
          },
          undefined,
        ],
      ]
    }
  >
    Welcome to BlueBase Framework!
  </Text>
</View>
`;
